### PR TITLE
cron fetch task usage data and populate audit table

### PR DESCRIPTION
**Pull request by bote795**
## Description

This cron runs on a daily basis to aggregate task audit data into day-wise task usage metrics for each deployment and ingests the same into the task usage table

## Related Issues

Related https://github.com/astronomer/issues/issues/5072

## Testing

CI

## Merging

0.31

### CODE REVIEW

File: charts/astronomer/values.yaml [hunk 1]
```diff
@@ -205,6 +205,14 @@ houston:
     # Default here is to run at 00:43 every night https://crontab.guru/#43_0_*_*_*
     schedule: "43 0 * * *"
 
+  # Populate daily task usage data
+  # This runs as a CronJob
+  populateHourlyTaskAuditMetrics:
+    # Enable check updates CronJob
+    enabled: true
```
Comment (rishkarajgi) [inline charts/astronomer/values.yaml:212]: we can remove this altogether and enable this cron whenever taskUsageMetrics feature flag is enabled

File: charts/astronomer/templates/houston/cronjobs/houston-populate-hourly-task-audit-metrics.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+########################################
+## Houston Populate Daily Task Metrics CronJob
+########################################
+{{- if .Values.houston.populateHourlyTaskAuditMetrics.enabled }}
```
Comment (rishkarajgi) [inline charts/astronomer/templates/houston/cronjobs/houston-populate-hourly-task-audit-metrics.yaml:4]: Change this to be based on feature flag instead
`{{- if .Values.global.taskUsageMetricsEnabled }}`

Comment (rishkarajgi) [reply charts/astronomer/templates/houston/cronjobs/houston-populate-hourly-task-audit-metrics.yaml:4]: Refer this PR - https://github.com/astronomer/astronomer/pull/1682

**Comment by rishkarajgi** [comment]
Please update the existing test file with test functions related to this cron 
[For reference](https://github.com/astronomer/astronomer/blob/master/tests/chart_tests/test_astronomer_houston_task_metrics.py)

File: charts/astronomer/values.yaml (hunk 1 above)
Comment (bote795) [reply charts/astronomer/values.yaml:212]: if i remove the block then have to put schedule somewhere else

**Comment by bote795** [comment]
ill branch out of ur pr and put a new pr

File: tests/chart_tests/test_astronomer_houston_task_metrics.py [hunk 3]
```diff
@@ -38,6 +36,37 @@ def test_astronomer_cleanup_task_usage_cron_feature_enabled(self, kube_version):
...
+        self, kube_version
+    ):
+        docs = render_chart(
+            kube_version=kube_version,
+            values={"global": {"taskUsageMetricsEnabled": False}},
+            show_only=[
+                "charts/astronomer/templates/houston/cronjobs/houston-populate-hourly-task-audit-metrics.yaml",
+            ],
+        )
+        assert len(docs) == 0
```
Comment (danielhoherd) [inline tests/chart_tests/test_astronomer_houston_task_metrics.py:49]: We should be asserting more things in this default test, such as contents that should exist and contents that should not exist.

File: tests/chart_tests/test_astronomer_houston_task_metrics.py [hunk 4]
```diff
@@ -38,6 +36,37 @@ def test_astronomer_cleanup_task_usage_cron_feature_enabled(self, kube_version):
...
+            ],
+        )
+
+        assert len(docs) == 1
+        assert docs[0]["kind"] == "CronJob"
+        assert (
+            docs[0]["metadata"]["name"]
+            == "release-name-houston-populate-hourly-task-audit-metrics"
+        )
+        assert docs[0]["spec"]["schedule"] == "57 * * * *"
```
Comment (danielhoherd) [inline tests/chart_tests/test_astronomer_houston_task_metrics.py:68]: We should have a test to verify that custom schedules work.

Comment (rishkarajgi) [reply tests/chart_tests/test_astronomer_houston_task_metrics.py:68]: done

File: tests/chart_tests/test_astronomer_houston_task_metrics.py (hunk 3 above)
Comment (rishkarajgi) [reply tests/chart_tests/test_astronomer_houston_task_metrics.py:49]: There won't be contents since doc won't render

**Review by danielhoherd — APPROVED** [review]

**Comment by danielhoherd** [comment]
LGTM. PR description needs to be completed.

**Pull request merged**